src/main/scala/io/iohk/ethereum/network/discovery/PeerDiscoveryManager.scala

```diff
@@ -106,7 +104,7 @@ class PeerDiscoveryManager(
     }
   }
 
-  private def getTcpPort: Int = nodeStatusHolder().serverStatus match {
+  private def getTcpPort: Int = nodeStatusHolder.get().serverStatus match {
```

Comment:
I don't know much about `AtomicReference ` but I'm curious if this `get()` returned in all calls in this class is the same or not? If first maybe we should think about extracting it to value to get rig of all this gets. If later skip this comment

Reply:
AtomicReference works as every Atomic* class in java. Under the hood there is a synchronisation on the reference. We shouldn't extract the value because we will lose atomicity.